tools/gcs-usage-recommender/README.md

```diff
@@ -0,0 +1,247 @@
+
+* Create and manage a BigQuery dataset in the project you’ve chosen
+  * bigquery.datasets.create
+  * bigquery.tables.create
+  * bigquery.tables.getData
+  * bigquery.tables.updateData
+
+And you will need permissions at the organization level to:
+
+* Create an organization-level audit log sink
+  * roles/logging.configWriter
+
+* Scan your inventory:
+  * storage.buckets.get
+  * storage.buckets.list
+  * storage.objects.get
+  * storage.objects.list
+  * resourcemanager.organizations.get
+  * resourcemanager.projects.get
+  * resourcemanager.projects.list
+
+You will need these permissions only during initial setup.
+
+<h3> 2.3 Preparing the audit log dataset</h3>
+Let’s walk through the steps necessary to create and populate the dataset.
+
+<h4> 2.3.1 Creating the dataset </h4>
+Let’s create the dataset inside the project you specified above. 
+<b>Note that this does not specify a partition expiration. Audit logs can get
+costly, so it is recommended to choose a default partition expiration for this
+dataset, such as 90 days.</b>
+
+````bq mk --location=${BQ_LOCATION} -d "${OUTPUT_PROJECT_ID}:${OUTPUT_DATASET_ID}"````
+
+<h4> 2.3.2 Create the audit log sink </h4>
+
+For GCS, we'll be enabling two types of [audit logs](https://cloud.google.com/storage/docs/audit-logs):
+* <b>Admin Activity logs</b>: Entries for operations that modify the 
+configuration or metadata of a project, bucket, or object.
+ 
+* <b>Data Access logs</b>: Entries for operations that modify objects or read a 
+project, bucket, or object. There are several sub-types of data access logs:
+    * <b>ADMIN_READ</b>: Entries for operations that read the configuration or 
+    metadata of a project, bucket, or object.
+    * <b>DATA_READ</b>: Entries for operations that read an object.
+    * <b>DATA_WRITE</b>: Entries for operations that create or modify an object.
+
+By default, only admin activity logs are enabled for organizations. However, data 
+access logs are not configured by default. This will require us to enable them in 
+the GCP console.
+
+To do this, go to the [console](https://pantheon.corp.google.com/iam-admin/audit?_ga=2.160269630.-2040617453.1540660549).
+![UI](images/audit_log_UI.png)
+Filter on Cloud Storage, and select ADMIN_READ, DATA_READ, AND DATA_WRITE. Then click save.
+
+Now, we will export the logs from Stackdriver to BigQuery.
+
+````bash
+gcloud logging sinks create gcs_usage \
+  bigquery.googleapis.com/projects/${AUDIT_LOG_PROJECT_ID}/datasets/${AUDIT_LOG_DATASET_ID} \
+  --log-filter='resource.type="gcs_bucket" AND
+  (protoPayload.methodName:"storage.buckets.delete" OR
+   protoPayload.methodName:"storage.buckets.create" OR
+   protoPayload.methodName:"storage.buckets.get" OR
+  protoPayload.methodName:"storage.objects.get" OR
+  protoPayload.methodName:"storage.objects.delete" OR
+  protoPayload.methodName:"storage.objects.create")' \
+  --organization=${ORG_ID} --include-children
+````
+

```

Comment:
In the spirit of cost optimization, consider recommending a logging exclusion filter to prevent the $0.50/ GB stackdriver logging ingest for these additional audit logs.
https://cloud.google.com/logging/docs/export/

Reply:
I filter only on the specific actions for GCS it requires (so I exclude out some ops, for things like service accounts, etc.) via the method names needed. Is there anything else that you'd recommend @jaketf ?

Reply:
What you're specifying here is a filter for the logging export (in this case to BQ).

However, as is all these GCS logs will be ingested into stackdriver logging in addition to BQ (which can add up). If the primary way the organization will look at GCS logs is through BQ (not Stack driver logging search / monitoring / alerting), then you should also exclude these logs from stackdriver logging so you're not paying $0.50/GB ingest for logs you're already ingesting to BQ (for much cheaper).

To do this you'd have to specify a separate filter to avoid these logs getting ingested into Stackdriver Logging via an [Exclusion](https://cloud.google.com/logging/docs/exclusions)
I would recommend mentioning exclusions as an optional step as there are valid reasons a user might want these logs in stackdriver logging as well (namely alerting).
I only brought this up because I think the spirit of this asset is cost savings and it's a good opportunity to remind customers that they can also save on their stackdriver bill by only ingesting to stackdriver logging what is actually giving them value in stackdriver (e.g. logs based metrics / alterting).

Reply:
Ah, okay sorry did not understand, very cool! Thanks LGTM